test/lightning_web/live/workflow_live/edit_test.exs

```diff
@@ -2357,6 +2357,311 @@ defmodule LightningWeb.WorkflowLive.EditTest do
 
       assert has_element?(view, "#ai-assistant-error", error_message)
     end
+
+    @tag email: "user@openfn.org"
+    test "assistant messages have properly styled links", %{
+      conn: conn,
+      project: project,
+      user: user,
+      workflow: %{jobs: [job_1 | _]} = workflow
+    } do
+      apollo_endpoint = "http://localhost:4001"
+
+      Mox.stub(Lightning.MockConfig, :apollo, fn
+        :endpoint -> apollo_endpoint
+        :openai_api_key -> "openai_api_key"
+      end)
+
+      Mox.stub(
+        Lightning.Tesla.Mock,
+        :call,
+        fn
+          %{method: :get, url: ^apollo_endpoint <> "/"}, _opts ->
+            {:ok, %Tesla.Env{status: 200}}
+
+          %{method: :post}, _opts ->
+            {:ok,
+             %Tesla.Env{
+               status: 200,
+               body: %{
+                 "history" => [
+                   %{"role" => "user", "content" => "Hello"},
+                   %{
+                     "role" => "assistant",
+                     "content" => """
+                     Here are some links:
+                     - [Apollo Repo](https://github.com/OpenFn/apollo)
+                     - Plain text
+                     - [Lightning Repo](https://github.com/OpenFn/lightning)
+                     """
+                   }
+                 ]
+               }
+             }}
+        end
+      )
+
+      insert(:chat_session, user: user, job: job_1)
+
+      {:ok, view, _html} =
+        live(
+          conn,
+          ~p"/projects/#{project.id}/w/#{workflow.id}?#{[v: workflow.lock_version, s: job_1.id, m: "expand"]}"
+        )
+
+      view
+      |> form("#ai-assistant-form")
+      |> render_submit(%{content: "Hello"})
+
+      html = render_async(view) |> Floki.parse_document!()
+
+      links = Floki.find(html, "a")
+
+      openfn_link =
+        links
+        |> Enum.find(
+          &(Floki.attribute(&1, "href") == ["https://github.com/OpenFn/apollo"])
+        )
+
+      assert openfn_link != nil
+
+      assert Floki.attribute(openfn_link, "class") == [
+               "text-primary-400 hover:text-primary-600"
+             ]
+
+      assert Floki.attribute(openfn_link, "target") == ["_blank"]
+
+      docs_link =
+        links
+        |> Enum.find(
+          &(Floki.attribute(&1, "href") == [
+              "https://github.com/OpenFn/lightning"
+            ])
+        )
+
+      assert docs_link != nil
+
+      assert Floki.attribute(docs_link, "class") == [
+               "text-primary-400 hover:text-primary-600"
+             ]
+
+      assert Floki.attribute(docs_link, "target") == ["_blank"]
+
+      assert Floki.find(html, "li")
+             |> Enum.any?(&(Floki.text(&1) == "\nPlain text  "))
```

Comment:
@elias-ba I've been thinking about this and I think instead of testing the chat component (at least this deeply), we should be testing the `apply_styles` function, perhaps that function becomes a function component that renders out html.

```
<.formatted_markdown content={message.content} />
```

And we do a function component test with different sets of markdown and test against that output?

Reply:
Thanks @stuartc, this is a great suggestion. I have implemented it.